src/ray/raylet/scheduling_resources.cc

```diff
@@ -135,15 +135,26 @@ double ResourceSet::GetNumCpus() const {
 
 const std::string ResourceSet::ToString() const {
   std::string return_string = "";
-  for (const auto &resource_pair : this->resource_capacity_) {
-    return_string +=
-        "{" + resource_pair.first + "," + std::to_string(resource_pair.second) + "}, ";
+
+  auto it = resource_capacity_.begin();
+
+  // Convert the first element to a string.
+  if (it != resource_capacity_.end()) {
+    return_string += "{" + it->first + "," + std::to_string(it->second) + "}";
+  }
+  it++;
+
+  // Add the remaining elements to the string (along with a comma).
+  while (it != resource_capacity_.end()) {
```

Comment:
why not: 
`for ( ; it != resource_capacity_.end() ; ++it )`
But I don't have a strong opinion on this.